### PR TITLE
Fix ripple effect on non-standard keyboards

### LINKED ISSUE #166: issue with ripple effect on Late 2016 RBS
Hi there, I've tried both razerCommander and polychromatic and both seem to have issues with the Ripple effect.  When I enable it in either, I see the following in `dmesg`:

```
[  252.870405] razerkbd: Command not supported. Start Marker: 05 id: ff Num Params: 46 Reserved: 03 Command: 0b Params: ff050015000000000000000000000000 .
```
repeatedly.

```sh
nick@nick-Blade-Stealth:~$ lsusb | grep '1532:'
Bus 001 Device 004: ID 1532:0220 Razer USA, Ltd
```

I have installed the drivers via:

```sh
nick@nick-Blade-Stealth:~$ sudo apt install python3-razer razer-kernel-modules-dkms razer-daemon razer-doc
```

### LINKED ISSUE #353: Razer Blade Stealth 13.3 (0x022d) ripple effect issue
[ripple_issue_pcaps.zip](https://github.com/terrycain/razer-drivers/files/1209348/ripple_issue_pcaps.zip)
The ripple effect does not work. As you suspected it uses the last of the single key setting the board had set. 

The ripple sngl required selecting effect / none twice before it actually went back to no effect so you may see that in the attached pcap.

BTW these pcaps are from the issue as seen on the linux side.

### PR DESCRIPTION
**Pull request by z3ntu**
Until now, ripple only worked on keyboards with a (6, 22) matrix.
Blade keyboards have different sizes (e.g. (6, 16) or (6, 25)) and they
should be supported.

~~This is not tested yet but it could work.~~

Fixes #166 and fixes #353

### CODE REVIEW

**Comment by KazWolfe** [comment]
No visible changes in ripple behavior on 2017 Blade Pro. Still has the initialization issue, and buttons are still off.

Not sure why, honestly. It feels like it should work, but it doesn't. Tried after reboot

**Comment by octylFractal** [comment]
Assuming I installed this right, I see no change either, with a 2017 Razer Blade. My keyboard turns into the image attached below. Using the following simple script works for other effects, but not ripple:
```
#!/usr/bin/env python3
import openrazer.client
dm = openrazer.client.DeviceManager()
if not dm.devices[0].fx.ripple_random():
    print('oh no')
```
Note: The script _does not_ print `oh no`, so the effect is still returning true.

Keyboard: keyboard image

**Comment by z3ntu** [comment]
@kenzierocks Can you add the line
```python
print("SET KEY COLOR : " + str(row) + " " +  str(col) + " " + colour)
```
under the line
```python
self._keyboard_grid.set_key_colour(row, col, colour)
```
and send me what gets printed?

**Comment by octylFractal** [comment]
It looks fine: https://gist.github.com/kenzierocks/42e271d4c47f820f6039848c6d43ee52

**Comment by z3ntu** [comment]
Hopefully https://github.com/openrazer/openrazer/blob/65c5d2d5cdd293939473aed2ee5a8c2a36acc0cc/daemon/openrazer_daemon/keyboard.py#L470 is the last culprit.

**Comment by z3ntu** [comment]
@kenzierocks Please try with the latest commit. If it doesn't work, add a
```python
print(len(payload))
```
after https://github.com/openrazer/openrazer/blob/3b465136b7f591d6fa8b657825a5cd1ece95ac6e/daemon/openrazer_daemon/misc/ripple_effect.py#L159

File: daemon/openrazer_daemon/misc/ripple_effect.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ def __init__(self, parent, device_number):
         self._shutdown = False
         self._active = False
 
+        self._matrixdims = self._parent._parent.MATRIX_DIMS
+
         self._keyboard_grid = KeyboardColour()
```
Comment (octylFractal) [inline daemon/openrazer_daemon/misc/ripple_effect.py]: It looks like you forgot to add the row + column arguments here

**Comment by octylFractal** [comment]
With the changes (plus a little local fix for the above error), it works beautifully :)

File: daemon/openrazer_daemon/misc/ripple_effect.py (hunk 1 above)
Comment (z3ntu) [reply daemon/openrazer_daemon/misc/ripple_effect.py]: Yep, sorry

**Comment by z3ntu** [comment]
Cool :) Can you check again if the state in this PR works (I amended the last commit for a nicer history)

**Comment by octylFractal** [comment]
Yes, the current branch tip is working as well.

**Comment by z3ntu** [comment]
Awesome, thanks for testing!

**Review by terricain — APPROVED** [review]

File: daemon/openrazer_daemon/misc/ripple_effect.py [hunk 2]
```diff
@@ -30,7 +30,9 @@ def __init__(self, parent, device_number):
         self._shutdown = False
         self._active = False
 
-        self._kerboard_grid = KeyboardColour()
+        self._matrixdims = self._parent._parent.MATRIX_DIMS
```
Comment (terricain) [inline daemon/openrazer_daemon/misc/ripple_effect.py]: I would of done something like `self._rows, self._cols = self._parent._parent.MATRIX_DIMS` but doesnt make much difference.

**Comment by terricain** [comment]
Merge and release when ready @z3ntu

**Comment by z3ntu** [comment]
@kenzierocks Please test once more, then I'll merge this :)

**Comment by octylFractal** [comment]
Latest commit is working perfectly :+1:

**Pull request merged**